qcfractal/alembic/versions/cd3d3e2915ce_add_indices_to_access_and_base_result.py

```diff
@@ -0,0 +1,34 @@
+"""Add indices to access and base_result
+
+Revision ID: cd3d3e2915ce
+Revises: c05d63683601
+Create Date: 2021-03-08 09:40:56.987985
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "cd3d3e2915ce"
+down_revision = "c05d63683601"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index("access_type", table_name="access_log")
+    op.create_index(op.f("ix_access_log_access_date"), "access_log", ["access_date"], unique=False)
+    op.create_index(op.f("ix_access_log_access_type"), "access_log", ["access_type"], unique=False)
+    op.create_index("ix_base_result_hash_index", "base_result", ["hash_index"], unique=True)
```

Comment:
We can't have the unique=True here although it's supposed to be unique. The reason is that it's not actually 100% unique. A rare case of duplicate hash can happen, although very tiny. So, we try to enforce this uniqueness in the python side to catch and handle this rare case, but not to leave it to Postgres.

Reply:
This is tricky. We aren't actually checking for that case (since that would be a bit expensive): https://github.com/MolSSI/QCFractal/blob/6d4fd08fcd84ac51262c75e18985c7d868df90e2/qcfractal/storage_sockets/sqlalchemy_socket.py#L723

If we wanted, we could still check for that case and warn the user, but still keep uniqueness in the database. I would expect it to be exceedingly rare, since molecule hashing uses sha1.

https://github.com/MolSSI/QCElemental/blob/ec9d754414824e129729f477e6077a14d859dc85/qcelemental/models/molecule.py#L758